### ISSUE TITLE
uniprot synonyms should be present for all ph7.3 forms

### ISSUE THREAD

**Issue by cmungall** [ebi-chebi/ChEBI#4519]
Needed for
- #4482

It looks like uniprot mappings may only have been added for leaf nodes.

The ph73 file has:

CHEBI:15356	CHEBI:35237	computation

where

* CHEBI:15356 https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:15356 cysteine
* CHEBI:35237 https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:35237 cysteine zwitterion

I would expect the secret uniprot biologist-friendly synonym to go on the zwitterion, but it's not present

**Comment by kaxelsen** [comment]
We only add UniProt names for compounds we use in UniProt, so as the flat cysteine zwitterion CHEBI:35237 is not used in UniProt, it has no UniProt name. Of that reason i most cases only the leaf node (L-cysteine zwitterion CHEBI:35235) has a UniProt name. 

If, for some reason, we had an enzyme that could use either or both L- and D- cysteine as substrates, and the assays couldn't distinguish between the two, then we would create a reaction using the flat cysteine (CHEBI:35235).

**Comment by cmungall** [comment]
Ideally CHEBI would provide a way to get the biologist-friendly terms for all protonated forms and we wouldn't be reliant on it being used in uniprot

**Comment by amalik01** [comment]
yes, i agree. One possibility is that ChEBI places a tag on each entry to indicate whether the structure is the major microspecies at pH 7.3 or not. I'm not entirely sure whether the major microspecies plugin in Marvin (https://docs.chemaxon.com/display/docs/calculators_major-microspecies-plugin.md) is suitable for bulk processing but can find out.

**Comment by redaschi** [comment]
We'd be super happy if CHEBI would take over the pH7.3 calculation :-) You can use Chemaxon for bulk processing. We currently do this for our pH7.3 mapping. But I had been told that you are moving away from Chemaxon? We will actually also soon start to evaluate alternatives.
NB: Not all our pH 7.3 forms are computed (as this is not possible), so you'd also have to curate some (Anne Morgat knows more about that).
